src/solver/expressions/visitors/AstGraphVisitor.cpp

```diff
@@ -0,0 +1,201 @@
+    dispatch(node->child());
+}
+
+void AstGraphVisitor::visit(const Nodes::PortFieldNode* node)
+{
+    int id = getNodeID(node);
+    emitNode(id,
+             "PF(" + node->getPortName() + "," + node->getFieldName() + ")",
+             "powderblue",
+             "invtriangle",
+             "filled, solid");
+}
+
+void AstGraphVisitor::visit(const Nodes::ComponentVariableNode* node)
+{
+    int id = getNodeID(node);
+    emitNode(id,
+             "CV(" + node->getComponentId() + "," + node->getComponentName() + ")",
+             "goldenrod",
+             "box",
+             "filled, solid");
+}
+
+void AstGraphVisitor::visit(const Nodes::ComponentParameterNode* node)
+{
+    int id = getNodeID(node);
+    emitNode(id,
+             "CP(" + node->getComponentId() + "," + node->getComponentName() + ")",
+             "palegreen",
+             "box",
+             "filled, solid");
+}
+
+std::string AstGraphVisitor::name() const
+{
+    return "AstGraphVisitor";
+}
+
+int AstGraphVisitor::getNodeID(const Nodes::Node* node)
+{
+    if (nodeIds_.find(node) == nodeIds_.end())
+    {
+        nodeIds_[node] = ++nodeCount_;
+    }
+    return nodeIds_[node];
+}
+
+void AstGraphVisitor::emitNode(int id,
+                               const std::string& label,
+                               const std::string& color,
+                               const std::string& shape,
+                               const std::string& style)
+{
+    *out_stream_ << "  " << id << " [label=\"" << label << "\", shape=\"" << shape << "\", style=\""
+                 << style << "\", color=\"" << color << "\"];\n";
+}
+
+// Process binary operation nodes like Add, Subtract, etc.
+void AstGraphVisitor::processBinaryOperation(const Nodes::BinaryNode* node,
+                                             const std::string& label,
+                                             const std::string& color,
+                                             const std::string& shape,
+                                             const std::string& style)
+{
+    int id = getNodeID(node);
+    emitNode(id, label, color, shape, style);
+
+    const Nodes::Node* left = node->left();
+    const Nodes::Node* right = node->right();
+
+    int leftId = getNodeID(left);
+    int rightId = getNodeID(right);
+
+    *out_stream_ << "  " << id << " -> " << leftId << ";\n";
+    *out_stream_ << "  " << id << " -> " << rightId << ";\n";
+
+    dispatch(left);
+    dispatch(right);
+}
+
+AstGraphVisitor::AstGraphVisitor(std::ostream* out_stream):
+    out_stream_(out_stream)
+{
+}
+
+AstGraphVisitorNotImplemented::AstGraphVisitorNotImplemented(const std::string& visitor,
+                                                             const std::string& node):
+    std::invalid_argument("Visitor" + visitor + " not implemented for node type " + node)
+{
+}
+
+void AstGraphVisitor::NewTreeGraph(const std::string& tree_name)
+{
+    *out_stream_ << "digraph " + tree_name + " {\n";
```

Comment:
Do we care about setting this ID ? It seems that it is only necessary to handle multiple graphs.